File: leetcode/go/2486.append-characters-to-string-to-make-subsequence.go
func appendCharacters(s string, t string) int {
    sp := 0
    tp := 0
    for (len(s) > sp && len(t) > tp) {
        if s[sp] == t[tp] {
            tp++
        }
        sp++
    }
    return (len(t)-tp)
}

File: leetcode/rust-leetcode/src/2486.append-characters-to-string-to-make-subsequence.rs
impl Solution {
    pub fn append_characters(s: String, t: String) -> i32 {
        let sb = s.into_bytes();
        let tb = t.into_bytes();
        let mut sp = 0;
        let mut tp = 0;
        while tp < tb.len() && sp < sb.len() {
            if sb[sp] == tb[tp] {
                tp += 1;
            }
            sp += 1;
        }
        (tb.len()-tp) as i32
    }
}


